framework/core/src/Post/CommentPost.php

```diff
@@ -145,10 +146,11 @@ public function getParsedContentAttribute()
      * Parse the content before it is saved to the database.
      *
      * @param string $value
+     * @param ServerRequestInterface $request
      */
-    public function setContentAttribute($value)
+    public function setContentAttribute($value, ServerRequestInterface $request = null)
     {
-        $this->attributes['content'] = $value ? static::$formatter->parse($value, $this) : null;
+        $this->attributes['content'] = $value ? static::$formatter->parse($value, $this, $request) : null;
     }
```

Comment:
I also don't like it very much, because of the limits it puts on us if we want to add more parameters later.

One solution could be to pass an array of options, one of which could be the request. The same array could then be passed to the formatter.

Another solution could be to ask the developers to use named parameters, and not make the order of the optional parameters part of the official API. But that requires PHP 8.0+

---

Speaking of the actual use case here, if the goal is to share the actor, I think extensions should be able to provide the actor without the need to build a request. What if an extension manually creates a post under a different user than the one currently logged in? What if it's created over a command line utility? Maybe it shouldn't even be called `actor`, but more something like `owner` to designate who wrote the content being saved in the current operation, no matter who actually performs the operation ?

Reply:
I agree with only passing the actor to `setContentAttribute` and using the post's author by default.

Reply:
Whilst I do agree with @clarkwinkelmann 's comment, and certainly passing the `owner` or `actor` arg would allow for greater felixibility going forward, the reasoning behind implementing it this way is to match how this is done already on the formatter's `render()` method.

https://github.com/flarum/framework/blob/main/framework/core/src/Formatter/Formatter.php#L107

That said, I'll see about getting this change implemented as soon as I can